src/telemetry/telemetry.c

```diff
@@ -248,7 +248,7 @@ build_version_request(const char *host, const char *path)
 }
 
 Connection *
-telemetry_connect(void)
+telemetry_connect(const char *host, const char *scheme)
```

Comment:
unnecessary but: should we do an `Assert` or check on `scheme` to make sure it is `<= 6` (`https` + `\0`)?  not sure what our standard for sanity checking is...

Reply:
I'd say no, it's better handled in `connection_connect`

Reply:
Sure, either place is fine with me, just meant in general.

Reply:
Hrm. I've thought about this a bit, and I'm still torn. On one hand, the check is not actually necessary since `getaddrinfo` should catch any ill-formed strings. Furthermore, the point of the connection library is to be as "general" as possible. On the other hand, we only use this connection library to send HTTP request atm, so maybe we should check that the service \in {'http', 'https'}. Anyone else want to weigh in @erimatnor @cevian @RobAtticus ?